### ISSUE TITLE
Audio output does not take sample rate into account

### ISSUE THREAD

**Issue by anisse** [Healthire/rust-game-bootstrap#2]
I’ve been playing with the audio output a bit, and it seems that it works by chance (at least on wasm, I didn't try on windows): it selects 96k samples/s 1 channel output mod in cpal, while the audio files (in ld48) are 48kHz 2 channels, so it works "magically".

But on Linux the cpal supported_output_config selection might give you a config with 384kHz (!), leaving the programmer wondering what she did wrong.

Anyway, I’m just opening this so you know and any potential user might know, too. The fix could be to do a more precise output config selection and maybe fail when no config matches what we want. Or to expose the output sample rate to the game code.

**Comment by HexyWitch** [comment]
Yeah I know about the issue and have just been too lazy to fix it, but it's something I want to handle properly. I think the right way to do it is to support any output sample rate and to resample in the mixer. Simple linear interpolation should be good enough for the purposes of this code I think.

Thank you for opening the issue! Sorry it's confusing. This is just a project that I use for my own little game jam games, and keep on github in case anyone wants to look at it. It's not *really* intended to be used by people.
